src/providers/routes.ts

```diff
@@ -0,0 +1,9 @@
+import express from 'express'
+import Controller from './controller'
+
+const router = express.Router()
+
+router.get('/connect/:provider', Controller.connect)
+router.get('/callback/:provider', Controller.callback)
```

Comment:
I had thought I couldn't do this due to how OAuth endpoints work, but I was wrong.

Will update.

Reply:
The previsou `/callback/:provider` route is still defined and working althoug deprecated. Here are the two locations it's used apparently, could be easy to update for the newly `/providers/:providerId/callback`.

https://github.com/verida/data-connector-server/blob/5f0cf9f82206802ae9eabf3030c1cff7910912c2/src/providers/index.ts#L12

https://github.com/verida/data-connector-server/blob/5f0cf9f82206802ae9eabf3030c1cff7910912c2/src/api/rest/v1/telegram/controller.ts#L117

Also the `/docs/Implementation.md` has to be updated, particularly the Connecting and Syncing Data sections

Reply:
This actually need to be updated because it's buggy.

`/callback/:provider` has a path param of `:provider` while `/providers/:providerId/callback` has `providerId. and the `Controller.callback` gets `providerId`, so all the callback coming from the old `/callback/:provider` end up failing

Reply:
Google doesn't like the redirect URL (`Error 400: redirect_uri_mismatch`) being `https://127.0.0.1:5021/providers/google/callback` maybe it's part of the configuration in the Google API/Console.

I reverted to the deprecated route and updated its definition to `/callback/providerId` and it's ok

Reply:
> The previous /callback/:provider route is still defined and working although deprecated

Yes, it's deprecated, not removed.

> Google doesn't like the redirect URL

This is why it's deprecated, not removed. The existing API connections need to be updated to use the new callback URL format, but if I do that it will break Don's work as he is working off a different branch.

I will clean this up once everything is merged and Don's work is merged. 

> /callback/:provider has a path param of :provider while /providers/:providerId/callback has providerId. and the Controller.callbackgetsproviderId, so all the callback coming from the old /callback/:provider` end up failing

Ah yes, thanks.

Should be fixed: https://github.com/verida/data-connector-server/commit/182a2790571c32879733b0ae4675b93232bd4a95